beluga/include/beluga/algorithm/estimation.hpp

```diff
@@ -171,7 +172,34 @@ std::pair<Sophus::SE2<Scalar>, Sophus::Matrix3<Scalar>> estimate(Poses&& poses,
 
 /// Computes mean and covariance Returns a pair consisting of the estimated mean pose and its covariance.
 /**
- * Given a range of 2D poses, computes the estimated pose by averaging the translation
+ * Given a range of poses, computes the estimated pose and the standard deviation by averaging the translation.
+ *
+ * \tparam Poses A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `std::vecotr<Scalar>`.
+ * \tparam Weights A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `Scalar`.
+ * \tparam Pose The pose value type of the given range.
+ * \param poses Range of 1D poses.
+ * \param weights Range of weights.
+ * \return The estimated pose and its standard deviation.
+ */
+template <
+    class Poses,
+    class Weights,
+    class Pose = ranges::range_value_t<Poses>,
+    typename = std::enable_if_t<std::is_arithmetic_v<Pose>>>
+std::pair<Pose, Pose> estimate(Poses&& poses, [[maybe_unused]] Weights&& weights) {
+  // TODO(alon): use the weights to calculate the weighted average.
+  auto accum_poses = ranges::accumulate(poses, 0.0);
+  Pose mean = accum_poses / (static_cast<Pose>(poses.size()));
+  auto variance = ranges::accumulate(poses, 0.0, [&mean](Pose init, Pose pose) { return init + pow(pose - mean, 2); });
```

Comment:
@Alondruck I suspect that `auto mean_deviation = pose - mean; return init + mean_deviation * mean_deviation;` will be more efficient than `pow`. I don't have data to back that up though.

Reply:
For single precision numbers (`float`), `std::pow(x, 2)` might be slower than `x * x` according to https://baptiste-wicht.com/posts/2017/09/cpp11-performance-tip-when-to-use-std-pow.html. At least in 2017.

Since this is a template, I'd say we stick to the `x * x` variant if we don't want to risk it.